### DISCUSSION TITLE
Disable built-in extensions

### DISCUSSION THREAD

**Discussion by the-sidner** [dedoc/scramble#418] [Ideas]
Description:

We have numerous resources in our project, all of which are suffixed with `Resource` (e.g., `ServerResource`, `NodeResource`). For our OpenAPI specification, we want to remove this `Resource` suffix. Currently, we achieve this by overwriting the `Dedoc\Scramble\Support\TypeToSchemaExtensions\JsonResourceTypeToSchema` class in our `composer.json` as follows:

```json
{
  "autoload": {
    "exclude-from-classmap": [
        "vendor/dedoc/scramble/src/Support/TypeToSchemaExtensions/JsonResourceTypeToSchema.php"
    ],
    "files": [
        "override/dedoc/scramble/src/Support/TypeToSchemaExtensions/JsonResourceTypeToSchema.php"
    ]
  }
}
```
Problem:

I am unsure how to replace the default `JsonResourceTypeToSchema` class without modifying the files in the `vendor` directory. When I create a new `TypeToSchema` extension, it runs concurrently with the built-in extension, which results in an unusable OpenAPI specification.

Request:

I would like to disable the default `JsonResourceTypeToSchema` extension and use my custom extension instead. How can I achieve this without directly modifying the vendor directory?

**Comment by romalytvynenko** [comment]
@the-sidner this is a very valid problem indeed. I will think about this. Any suggestions about most DX-friendly API are welcome!
